src/Accelerators/NNPA/Compiler/NNPACompilerOptions.cpp

```diff
@@ -100,4 +100,10 @@ llvm::cl::opt<bool> nnpaEnableSaturation("nnpa-saturation",
                    "Default is false."),
     llvm::cl::init(false), llvm::cl::cat(OnnxMlirCommonOptions));
 
+llvm::cl::opt<bool> nnpaDelayStickifiedConstGen(
+    "nnpa-delay-stickified-const-generation",
+    llvm::cl::desc(
+        "Generate stickified constants when storing constants to file. This "
+        "will be able to reduce peak memory usage. Default is true."),
```

Comment:
I see you always check `if (storeConstantsToFile && nnpaDelayStickifiedConstGen)`. So, it seems this option only works when `store-constant-to-file` is enabled. I recommend to turn it off at the beginning of the compilation if `store-constant-to-file` is off, then you need to check `if (nnpaDelayStickifiedConstGen)` only.

By that way you don't need to check

Reply:
I removed this option. Thanks.